[quant][graphmode][fx] Enable fuse handler for sequence of 3 ops

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #69882
* #69778
* __->__ #69658

Summary:
This PR enables fuse handler for sequence of three ops, and merges all fuse handlers into one

TODO: we can also move this to backend_config_dict folder

Test Plan:
regression fusion test
```
python test/test_quantization.py TestFuseFx
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D32974907](https://our.internmc.facebook.com/intern/diff/D32974907)

jerryzh168 committed:
[quant][graphmode][fx] Enable fuse handler for sequence of 3 ops

Summary:
This PR enables fuse handler for sequence of three ops, and merges all fuse handlers into one

TODO: we can also move this to backend_config_dict folder

Test Plan:
regression fusion test
```
python test/test_quantization.py TestFuseFx
```

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 commented:
@jerryzh168 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32974907).

jerryzh168 committed:
Update on "[quant][graphmode][fx] Enable fuse handler for sequence of 3 ops"


Summary:
This PR enables fuse handler for sequence of three ops, and merges all fuse handlers into one

TODO: we can also move this to backend_config_dict folder

Test Plan:
regression fusion test
```
python test/test_quantization.py TestFuseFx
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D32974907](https://our.internmc.facebook.com/intern/diff/D32974907)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][fx] Enable fuse handler for sequence of 3 ops"


Summary:
This PR enables fuse handler for sequence of three ops, and merges all fuse handlers into one

TODO: we can also move this to backend_config_dict folder

Test Plan:
regression fusion test
```
python test/test_quantization.py TestFuseFx
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D32974907](https://our.internmc.facebook.com/intern/diff/D32974907)

[ghstack-poisoned]

jerryzh168 commented:
@jerryzh168 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32974907).

vkuzo commented on torch/ao/quantization/fx/fusion_patterns.py:
```diff
@@ -136,26 +72,41 @@ def fuse(self, quantizer: QuantizerCls,
-                relu.training = root_module.training
-                return relu
-            return MatchAllNode
+        def get_modules(pattern, modules):
```
can we add a quick docblock with an example of what this is doing?

jerryzh168 committed:
Update on "[quant][graphmode][fx] Enable fuse handler for sequence of 3 ops"


Summary:
This PR enables fuse handler for sequence of three ops, and merges all fuse handlers into one

TODO: we can also move this to backend_config_dict folder

Test Plan:
regression fusion test
```
python test/test_quantization.py TestFuseFx
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D32974907](https://our.internmc.facebook.com/intern/diff/D32974907)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][fx] Enable fuse handler for sequence of 3 ops"


Summary:
This PR enables fuse handler for sequence of three ops, and merges all fuse handlers into one

TODO: we can also move this to backend_config_dict folder

Test Plan:
regression fusion test
```
python test/test_quantization.py TestFuseFx
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D32974907](https://our.internmc.facebook.com/intern/diff/D32974907)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][fx] Enable fuse handler for sequence of 3 ops"


Summary:
This PR enables fuse handler for sequence of three ops, and merges all fuse handlers into one

TODO: we can also move this to backend_config_dict folder

Test Plan:
regression fusion test
```
python test/test_quantization.py TestFuseFx
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D32974907](https://our.internmc.facebook.com/intern/diff/D32974907)

[ghstack-poisoned]

jerryzh168 commented:
@jerryzh168 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32974907).

jerryzh168 committed:
Update on "[quant][graphmode][fx] Enable fuse handler for sequence of 3 ops"


Summary:
This PR enables fuse handler for sequence of three ops, and merges all fuse handlers into one

TODO: we can also move this to backend_config_dict folder

Test Plan:
regression fusion test
```
python test/test_quantization.py TestFuseFx
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D32974907](https://our.internmc.facebook.com/intern/diff/D32974907)

[ghstack-poisoned]

jerryzh168 commented:
@jerryzh168 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32974907).

jerryzh168 committed:
Update on "[quant][graphmode][fx] Enable fuse handler for sequence of 3 ops"


Summary:
This PR enables fuse handler for sequence of three ops, and merges all fuse handlers into one

TODO: we can also move this to backend_config_dict folder

Test Plan:
regression fusion test
```
python test/test_quantization.py TestFuseFx
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D32974907](https://our.internmc.facebook.com/intern/diff/D32974907)

[ghstack-poisoned]
